internal/agent/http/poller.go

```diff
@@ -0,0 +1,89 @@
+package http
+
+import (
+	"log"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/Xacor/go-metrics/internal/agent/metric"
+)
+
+type Poller struct {
+	pollInterval   int
+	reportInterval int
+	address        string
+	metrics        *metric.Metrics
+	client         *http.Client
+}
+
+func NewPoller(pollInterval, reportInterval int, address string) *Poller {
+	return &Poller{
+		pollInterval:   pollInterval,
+		reportInterval: reportInterval,
+		metrics:        metric.NewMetrics(),
+		address:        "http://" + address,
+		client:         &http.Client{},
+	}
+}
+
+func (p *Poller) Run() {
+	log.Println("poller started")
+	for i := 0; ; i++ {
+		log.Println("poller tick:", i)
+		time.Sleep(time.Second * 1)
+		if i%p.pollInterval == 0 {
+			p.metrics.Update()
+		}
+		if i%p.reportInterval == 0 {
+			err := p.SendRequests()
+			if err != nil {

```

Comment:
(Не ошибка реализации) Когда возвращается только ошибка ещё используют конструкцию.
```suggestion
			if err := p.SendRequests(); err!=nil{}

```